src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -492,6 +492,21 @@ private static void ParseArgumentsIntoList(string arguments, List<string> result
                 // it contains spaces.
                 if (c == '"')
                 {
+                    // While not in quotes, and currentArgument is empty,
```

Comment:
While this may patch this particular problem, it doesn't address the root cause - that the code was using "currentArgument.Length &gt; 0" as a flawed proxy for the real question of "have we seen at least one non-whitespace character since the last time we saw a whitespace character?"

I think this code also has other problems regarding the double double quote sequence.

Since the purpose of this code is to match the tokenizing that happens on Windows when it sets up the args array passed to main(), I think it'd be best just to reuse that code and get the canonical behavior for sure:

https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/src/System/Runtime/CommandLine.Windows.cs#L44

You can drop the ScanArgument0() part and other stuff relating to parsing argv[0] since that's handled separately.